elasticdl/python/common/model_handler.py

```diff
@@ -44,6 +44,16 @@ def _convert_embedding_table_to_numpy_array(embedding_table, embedding_shape):
     return embedding_weights
 
 
+def _need_partition_embedding(layer):
+    """The embedding layer will be partitioned on multiple
+    PS instances if the memory of the layer.train_weights is
+    bigger than 2MB.
+    """
+    EMBEDDING_SIZE_THRESHOLD_FOR_PARTITION = 2 * 1024 * 1024  # 2MB
+    weights_memory = layer.input_dim * layer.output_dim * 8
```

Comment:
If the weight type is float32, it should be `layer.input_dim * layer.output_dim * 4`, since one float32 occupy 4 bytes.

Reply:
Yes, i will check the dtype of embedding weights.